internal/driver/metadata.go

```diff
@@ -266,5 +279,37 @@ func getSupportedFrameRateFormats(d *usbdevice.Device) (interface{}, error) {
 		}
 		r.FrameRateFormats = append(r.FrameRateFormats, format)
 	}
-	return r, nil
+	resultMap := make(map[string]result)
+	resultMap[d.Name()] = r
+	return resultMap, nil
+}
+
+func GetFrameRate(d *usbdevice.Device) (interface{}, error) {
+	streamParam, err := d.GetStreamParam()
+	if err != nil {
+		return v4l2.Fract{}, err
```

Comment:
should probably be `nil, err` if possible. the return type here does not match below